pkg/ottl/e2e/e2e_test.go

```diff
@@ -1381,6 +1416,21 @@ func constructSpanTransformContext() ottlspan.TransformContext {
 	return ottlspan.NewTransformContext(td, scope, resource, ptrace.NewScopeSpans(), ptrace.NewResourceSpans())
 }
 
+func constructSpanEventTransformContext() ottlspanevent.TransformContext {
+	resource := pcommon.NewResource()
+
+	scope := pcommon.NewInstrumentationScope()
+	scope.SetName("scope")
+
+	span := ptrace.NewSpan()
+	fillSpanOne(span)
+
+	ev1 := span.Events().AppendEmpty()
```

Comment:
nit: I'd add one more span event here and test it by checking all elements `event_index` values. I know It's not really necessary to ensure the current logic is working, but given the zero value of integers is `0`, it would catch future bugs like not setting the `event_index` value on the `TransformContext`, and/or filling it out of order.

Reply:
this should be resolved now as we pass the index via the option